packages/library/src/transaction-confirmation-strategy-racer.ts

```diff
@@ -0,0 +1,40 @@
+import { Commitment } from '@solana/rpc-core';
+import { TransactionSignature } from '@solana/transactions';
+
+import { createRecentSignatureConfirmationPromiseFactory } from './transaction-confirmation-strategy-recent-signature';
+
+export interface BaseTransactionConfirmationStrategyConfig {
+    abortSignal: AbortSignal;
+    commitment: Commitment;
+    getRecentSignatureConfirmationPromise: ReturnType<typeof createRecentSignatureConfirmationPromiseFactory>;
+}
+
+export async function raceStrategies<TConfig extends BaseTransactionConfirmationStrategyConfig>(
+    signature: TransactionSignature,
+    config: TConfig,
+    getSpecificStrategiesForRace: (config: TConfig) => readonly Promise<unknown>[]
+) {
+    const { abortSignal: callerAbortSignal, commitment, getRecentSignatureConfirmationPromise } = config;
```

Comment:
Is there a reason we have them pass just the abort signal and not the whole `AbortController`? It seems like that would make this simpler since we wouldn't need to create our own controller and hook the events together.

Reply:
Ohhhh… interesting. That does work for ‘internal-y’ code, doesn't it.

Reply:
Oh, actually, the reason is because we don't _own_ it. All we have in this chain of calls is the `AbortSignal` supplied by the developer. They don't (and shouldn't) give us their `AbortController`.